File: src/test/java/com/example/scaffold/PingUseCase.java
package com.example.scaffold;

import com.example.frameworks.application.core.UseCase;
import com.example.scaffold.frameworks.valueobjects.payload.RequestPayload;
import com.example.scaffold.frameworks.valueobjects.payload.ResponsePayload;

import static com.example.scaffold.frameworks.valueobjects.payload.StringFieldType.asString;

@UseCase
public class PingUseCase {
    public ResponsePayload execute(RequestPayload payload) {
        return new ResponsePayload().append("ping", payload.get("data", asString()));
    }
}

File: src/test/kotlin/com/example/scaffold/PingUseCase.kt
package com.example.scaffold

import com.example.frameworks.application.core.UseCase
import com.example.frameworks.domain.core.Payload
import com.example.frameworks.domain.core.StringFieldType.Companion.asString

@UseCase
class PingUseCase {
    fun execute(payload: Payload): String {
        return payload.get("data", asString())
    }
}
